msal/src/Microsoft.Identity.Client/IMsalHttpClientFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System.Net.Http;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Interface for managing the HttpClient used within MSAL.  This can be set within the MsalConfigurationBuilder
+    /// as part of constructing a PublicClientApplication or ConfidentialClientApplication object.
+    /// </summary>
+    public interface IMsalHttpClientFactory

```

Comment:
I agree with @henrik-me that we should not prefix things with "MSAL" but in this case I think it might be justified because IHttpClientFactory also exists in ASP.net and would lead to confusion. Alternatively, we could expose a lambda.